### PR TITLE
Fix scf.forall to scf.parallel conversion

### PR DESCRIPTION
**Pull request by chelini**
Upstream converts `scf.forall` to `scf.parallel`, which is, in turn, lowered to an `scf.for` and then to structured control flow in `convert-scf-to-cf`. This lowering is atomic, thus we need to materialize `scf.parallel` before `convert-scf-to-cf` if we want to map to `omp.parallel`. The ideal solution would be to decouple the lowering from `scf.forall` to `scf.parallel` from upstream: `convert-scf-to-cf`.

fixes: #734

### CODE REVIEW

File: test/Integration/openmp/tpp-run-parallel.mlir [hunk 1]
```diff
@@ -1,6 +1,8 @@
-// RUN: tpp-run %s -def-parallel=0 -e entry -entry-point-result=void -print-mlir=late -print 2>&1 | \
+// RUN: tpp-opt %s -default-tpp-passes | \
```
Comment (rengolin) [inline test/Integration/openmp/tpp-run-parallel.mlir:1]: Why do you need to pipe `tpp-opt`?

**Comment by rengolin** [comment]
Some weird crashes on the tests, did you not see this on your machine?

File: test/Integration/openmp/tpp-run-parallel.mlir (hunk 1 above)
Comment (chelini) [reply test/Integration/openmp/tpp-run-parallel.mlir:1]: We need to make sure we get from scf.forall to scf.parallel. Previously we were only lowering: `scf.parallel` using tpp-run this did not catch the regression.

**Comment by chelini** [comment]
> Some weird crashes on the tests, did you not see this on your machine?

nope.

File: test/Integration/openmp/tpp-run-parallel.mlir (hunk 1 above)
Comment (rengolin) [reply test/Integration/openmp/tpp-run-parallel.mlir:1]: That's what I want to understand. both `tpp-opt` and `tpp-run` need to converge to the same IR, especially when both are running the default TPP passes. If they don't converge, then there is a pipeline problem.

Comment (chelini) [reply test/Integration/openmp/tpp-run-parallel.mlir]: there was not need for `tpp-opt`.

**Review by rengolin — APPROVED** [review]

**Pull request merged**